GUI: move game settings out of Engine tab

Die4Ever opened the pull request:
I think these options are a bit too hidden for most users to find. Moving them to the Game tab ensures everyone will see the options immediately upon adding a game. This is great for an option like "Easier AI" for The 11th Hour, where not knowing about this option could cause a lot of needless frustration for the player.

Currently there's extra spacing that I don't know how to get rid of, I'm sure it's something very simple. But at least it works!

![image](https://user-images.githubusercontent.com/30947252/144814585-8c6734fa-0d74-47a0-a99c-368757d8310a.png)

Die4Ever commented on gui/editgamedialog.cpp:
```diff
@@ -184,25 +184,23 @@ EditGameDialog::EditGameDialog(const Common::String &domain)
 		if (enginePlugin) {
 			enginePlugin->get<MetaEngine>().registerDefaultSettings(_domain);
-			_engineOptions = enginePlugin->get<MetaEngine>().buildEngineOptionsWidgetDynamic(tab, "GameOptions_Engine.Container", _domain);
+			_engineOptions = enginePlugin->get<MetaEngine>().buildEngineOptionsWidgetDynamic(tab, "GameOptions_Game.Container", _domain);
```
we could even keep buildEngineOptionsWidgetDynamic using the separate Engine tab if we want, or make an additional function for that

sev- commented:
This will not work on a low resolution like 320x200. You could, however, potentially, split this code depending on the resolution.

Die4Ever commented:
> This will not work on a low resolution like 320x200. You could, however, potentially, split this code depending on the resolution.

now with scrollbar!

![image](https://user-images.githubusercontent.com/30947252/144988359-164bd76c-7b09-4ca2-94c0-bbbb4692ff44.png)

and I fixed the spacing issue

![image](https://user-images.githubusercontent.com/30947252/144988380-6c3a8382-57c7-45bb-bf43-ab8ef7c6ea57.png)

Die4Ever commented on gui/themes/common/lowres_layout.stx:
```diff
@@ -1266,6 +1266,12 @@
+	
+	<dialog name = 'GameOptions_Game_Container' overlays ='Dialog.GameOptions_Game.Container' inset='0'>
 		<layout type = 'vertical' padding = '8, 8, 8, 8'>
 			<layout type = 'horizontal' padding = '0, 0, 0, 0' spacing = '6' align = 'center'>
```
now that we have a scrollbar, we could remove these horizontal layouts from lowres if we wanted to

sev- commented:
Looks good. However, 2 things are missing:
 - Bump theme version (THEMERC file and `gui/ThemeEngine.h` file)
 - Update classic theme

Die4Ever commented:
this is probably superseded by #3594 although that one is a different approach and more complex

bluegr commented:
> this is probably superseded by #3594 although that one is a different approach and more complex

You mean one part of this pull request, right? Cause one pull request adds tab scrolling and the other moves options to another tab

Die4Ever commented:
> > this is probably superseded by #3594 although that one is a different approach and more complex
> 
> You mean one part of this pull request, right? Cause one pull request adds tab scrolling and the other moves options to another tab

No, the other pull request does that too, it just doesn't require as large of a change in the themes files to accomplish it

bluegr commented:
Closing this one in favor of #3594, which provides more flexibility
